From b0690b5a799b876e1f8e1b1478fd28b10fdf5a3e Mon Sep 17 00:00:00 2001
From: "pj.chen" <chen.pj@inventec.com>
Date: Mon, 13 Dec 2021 02:12:10 +0000
Subject: [PATCH] Bug 713 - [SW][Common][phosphor-nvme] - Remove the dbus
 sensor create function

Symptom/Reason:
    Since we were using intel-ipmi-oem for SDR function,
    the dbus interface should follow the dbus-sensors's way to create the dbus service.
    The "phosphor-nvme" using the function created by "phosphor-dbus-interface",
    which would not have "org.freedesktop.DBus.ObjectManager -> GetManagedObjects" at root "/".

    So here we should block the dbus creation in phosphor-nvme,
    if there's the need for the thermal sensor on nvme, would use dbus-sensors to create.

Root Cause:
    N/A

Solution/Change:
    Add flag CREATE_TEMPERATURE_SENSORS to seperate the sensor creation.

Entry Test:

sysadmin@starscream:~# busctl tree xyz.openbmc_project.nvme.manager
`-/xyz
  `-/xyz/openbmc_project
    `-/xyz/openbmc_project/sensors
      `-/xyz/openbmc_project/sensors/temperature
---
 nvme_manager.cpp | 6 ++++++
 nvme_manager.hpp | 1 +
 2 files changed, 7 insertions(+)

diff --git a/nvme_manager.cpp b/nvme_manager.cpp
index 02bc819..271d109 100644
--- a/nvme_manager.cpp
+++ b/nvme_manager.cpp
@@ -528,6 +528,8 @@ void Nvme::readNvmeData(NVMeConfig& config)
 
     // get NVMe information through i2c by busID.
     auto success = getNVMeInfobyBusID(config.busID, nvmeData);
+
+#if CREATE_TEMPERATURE_SENSORS
     auto iter = nvmes.find(config.index);
 
     // can not find. create dbus
@@ -561,6 +563,10 @@ void Nvme::readNvmeData(NVMeConfig& config)
         iter->second->checkSensorThreshold();
         setLEDsStatus(config, success, nvmeData);
     }
+#else
+    setNvmeInventoryProperties(true, nvmeData, inventoryPath);
+    setLEDsStatus(config, success, nvmeData);
+#endif
 }
 
 /** @brief Monitor NVMe drives every one second  */
diff --git a/nvme_manager.hpp b/nvme_manager.hpp
index 97ef204..ffe0a8b 100644
--- a/nvme_manager.hpp
+++ b/nvme_manager.hpp
@@ -17,6 +17,7 @@ namespace phosphor
 {
 namespace nvme
 {
+#define CREATE_TEMPERATURE_SENSORS 0
 
 /** @class Nvme
  *  @brief Nvme manager implementation.
-- 
2.17.1

